jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -1053,6 +1056,7 @@ public boolean handle(Request request, Response response, Callback callback) thr
             return true;
 
         // Past this point we are calling the downstream handler in scope.
+        Context lastContext = getCurrentContext();
```

Comment:
@sbordet this is an unrelated change that was causing flaky tests as it was leaving the ThreadLocal context set on the main test thread!